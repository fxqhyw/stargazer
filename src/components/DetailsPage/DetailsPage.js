import React, { useEffect } from 'react'
import { useSelector } from 'react-redux'
import { useHistory, useParams } from 'react-router-dom'

const DetailsPage = () => {
  const { id } = useParams()
  const history = useHistory()
  const repository  = useSelector(state => state.repositories.data[`${id}`])


  useEffect(() => {
    if (!repository) {
      history.push('/')
    }
  }, [repository])

  return(
    <div>Details page {id}</div>
  )
}

export default DetailsPage