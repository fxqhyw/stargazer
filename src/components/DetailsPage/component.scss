.details-page {
  margin: 10rem auto 0 auto;
  max-width: 30%;

  .details-head {
    font-size: 20px;
    margin-bottom: 4rem;
  }

  .stats-col {
    width: 50%;
    display: inline-grid;

    .stats-head {
      font-size: 18px;
      margin-bottom: .5rem;
    }
  }

  .no-repo {
    margin-bottom: 50px;
  }
}
