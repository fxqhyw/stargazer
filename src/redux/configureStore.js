import { createStore, applyMiddleware } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly'
import { createLogicMiddleware } from 'redux-logic'
import Reducer from './reducers'
import rootLogic from './logic/'
import axios from 'axios'

const initialState = {
  repositories: {
    data: {},
    error: null
  }
}

const logicDeps = {
    httpClient: axios
}

const logicMiddleware = createLogicMiddleware(rootLogic, logicDeps);


export default function configureStore() {
  return createStore(
    Reducer,
    initialState,
    composeWithDevTools(
      applyMiddleware(logicMiddleware),
    )
  )
}

