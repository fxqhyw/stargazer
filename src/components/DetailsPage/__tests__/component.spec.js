import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { mount } from 'enzyme'
import DetailsPage from '../component'

const mockStore = configureStore([])

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
  useParams: () => ({
    id: "10270250",
  })
}))
 
describe('DetailsPage', () => {
  let store;
 
  beforeEach(() => {
    store = mockStore({
      repositories: {
        data: {
          "10270250": {
            id: 10270250,
            full_name: "facebook/react",
            stargazers_count: 154386,
            subscribers_count: 6678,
            forks_count: 30270,
            clone_url: "https://github.com/facebook/react.git",
            languages_url: "https://api.github.com/repos/facebook/react/languages",
            languages: {
              JavaScript: "94.9%",
              HTML: "2.2%",
              CSS: "1.2%",
              "C++": "0.8%",
              TypeScript: "0.4%",
              CoffeeScript: "0.3%",
              C: "0.1%",
              Shell: "0.1%"
            }
          }
        }
      },
    })

    store.dispatch = jest.fn()
  })

  it('should render correctly', () => {
    const component = mount(
      <Provider store={store}>
        <DetailsPage />
      </Provider>
    )
    expect(component).toMatchSnapshot()
    component.unmount()
  })
});
