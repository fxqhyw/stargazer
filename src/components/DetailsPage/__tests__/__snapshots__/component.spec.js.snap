// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage should render correctly 1`] = `
<div
  className="details-page"
>
  <div
    className="details-head"
  >
    <div
      className="stats-col"
    >
      facebook/react
    </div>
    <div
      className="stats-col"
    >
      <a
        href="https://github.com/facebook/react.git"
      >
        Clone
      </a>
    </div>
  </div>
  <div>
    <div
      className="stats-col"
    >
      <div
        className="stats-head"
      >
        Stats:
      </div>
      <div>
        Watchers: 
        6678
      </div>
      <div>
        Stars: 
        154386
      </div>
      <div>
        Forks: 
        30270
      </div>
    </div>
    <div
      className="stats-col"
    >
      <div
        className="stats-head"
      >
        Languages:
      </div>
      <div
        key="JavaScript"
      >
        JavaScript
        : 
        94.9%
      </div>
      <div
        key="HTML"
      >
        HTML
        : 
        2.2%
      </div>
      <div
        key="CSS"
      >
        CSS
        : 
        1.2%
      </div>
      <div
        key="C++"
      >
        C++
        : 
        0.8%
      </div>
      <div
        key="TypeScript"
      >
        TypeScript
        : 
        0.4%
      </div>
      <div
        key="CoffeeScript"
      >
        CoffeeScript
        : 
        0.3%
      </div>
      <div
        key="C"
      >
        C
        : 
        0.1%
      </div>
      <div
        key="Shell"
      >
        Shell
        : 
        0.1%
      </div>
    </div>
  </div>
  <Button
    active={false}
    disabled={false}
    onClick={[Function]}
    size="sm"
    variant="primary"
  >
    Go back
  </Button>
</div>
`;
